crates/swc_error_reporters/src/lib.rs

```diff
@@ -175,9 +178,19 @@ impl Emitter for PrettyEmitter {
             children,
         };
 
+        let mut format_result = String::new();
+
         self.reporter
-            .render_report(&mut self.wr, &diagnostic)
+            .render_report(&mut format_result, &diagnostic)
```

Comment:
I think this may be fine without an additional option, considering we were already formating as string (rendering using `miette`) error messages, and `json` support is out of the scope of this PR anyway.